brewtils/test/comparable.py

```diff
@@ -330,6 +330,34 @@ def assert_job_equal(obj1, obj2, do_raise=False):
     )
 
 
+def assert_job_ids_equal(dict1, dict2, do_raise=False):
```

Comment:
I got exhausted just looking at this. This being necessary feels like a huge red flag to me, but given that it's inline with the existing stuff, just carry on I guess.